packages/core/test/financial-templates/common/financial-product-libraries/long-short-pair-libraries/BinaryOptionLongShortPairFinancialProductLibrary.js

```diff
@@ -1,79 +1,108 @@
+const hre = require("hardhat");
+const { runDefaultFixture } = require("@uma/common");
+const { getContract } = hre;
 const { didContractThrow, ZERO_ADDRESS } = require("@uma/common");
 const { assert } = require("chai");
 
 // Tested Contract
-const BinaryOptionLongShortPairFinancialProductLibrary = artifacts.require(
+const BinaryOptionLongShortPairFinancialProductLibrary = getContract(
   "BinaryOptionLongShortPairFinancialProductLibrary"
 );
 
 // helper contracts. To test LSP libraries we simply need a financial contract with an `expirationTimestamp` method.
-const ExpiringContractMock = artifacts.require("ExpiringMultiPartyMock");
+const ExpiringContractMock = getContract("ExpiringMultiPartyMock");
 
 const { toWei, toBN, utf8ToHex } = web3.utils;
 const strikePrice = toBN(toWei("3000"));
 
-contract("BinaryOptionLongShortPairFinancialProductLibrary", function () {
+describe("BinaryOptionLongShortPairFinancialProductLibrary", function () {
   let binaryLSPFPL;
   let expiringContractMock;
+  let accounts;
+
+  before(async () => {
+    await runDefaultFixture(hre);
+    accounts = await hre.web3.eth.getAccounts();
+  });
 
   beforeEach(async () => {
-    binaryLSPFPL = await BinaryOptionLongShortPairFinancialProductLibrary.new();
+    binaryLSPFPL = await BinaryOptionLongShortPairFinancialProductLibrary.new().send({ from: accounts[0] });
     expiringContractMock = await ExpiringContractMock.new(
       ZERO_ADDRESS, // _financialProductLibraryAddress
       "1000000", // _expirationTimestamp
       { rawValue: toWei("1.5") }, // _collateralRequirement
       utf8ToHex("TEST_IDENTIFIER"), // _priceIdentifier
       ZERO_ADDRESS // _timerAddress
-    );
+    ).send({ from: accounts[0] });
   });
   describe("Long Short Pair Parameterization", () => {
     it("Can set and fetch valid values", async () => {
-      await binaryLSPFPL.setLongShortPairParameters(expiringContractMock.address, strikePrice);
+      await binaryLSPFPL.methods
+        .setLongShortPairParameters(expiringContractMock.options.address, strikePrice)
+        .send({ from: accounts[0] });
 
-      const setParams = await binaryLSPFPL.longShortPairParameters(expiringContractMock.address);
+      const setParams = await binaryLSPFPL.methods.longShortPairParameters(expiringContractMock.options.address).call();
       assert.isTrue(setParams.isSet);
       assert.equal(setParams.strikePrice.toString(), strikePrice);
     });
     it("Can not re-use existing LSP contract address", async () => {
-      await binaryLSPFPL.setLongShortPairParameters(expiringContractMock.address, strikePrice);
+      await binaryLSPFPL.methods
+        .setLongShortPairParameters(expiringContractMock.options.address, strikePrice)
+        .send({ from: accounts[0] });
 
       // Second attempt should revert.
       assert(
-        await didContractThrow(binaryLSPFPL.setLongShortPairParameters(expiringContractMock.address, strikePrice))
+        await didContractThrow(
```

Comment:
Seeing all these changes in clude the `{ from:accounts[0] }` I'm going to assume this is a strict requirement for this. is there no way to tell the contract instance to default to a particular account? it would make these kinds of sets cleaner. I guess it is more explicitly to define every "from" address every time a transaction is sent.

Reply:
main downside is this becomes pretty repetitive and the syntax is more verbose on every function call

Reply:
I think you can do it (because web3 instances have a default account field), but you'd need to modify web3 before the tests start using it (although I'm not 100% sure how you do this) since this doesn't happen by default. We could make a plug-in to do it if we want to have that for future tests.

Reply:
I do think explicitness is nice. In the tests, we are pretty inconsistent about denoting the account. Sometimes we define an owner and specifically call things from the owner. Other times, we rely on the truffle default.

I can try to include a default in web3 in a later PR and we can use whatever pattern we want for new files/tests!

Reply:
I also favor explicitness its more readable